lib/datadog/core/configuration/option.rb

```diff
@@ -98,6 +98,20 @@ def coerce_env_variable(value)
           return context_exec(value, &@definition.env_parser) if @definition.env_parser
 
           case @definition.type
+          when :hash
+            values = value.split(',') # By default we only want to support comma separated strings
+
+            values.map! do |v|
+              v.gsub!(/\A[\s,]*|[\s,]*\Z/, '')
+
+              v.empty? ? nil : v
+            end
+
+            values.compact!
+            values.each.with_object({}) do |v, hash|
+              pair = v.split(':', 2)
+              hash[pair[0]] = pair[1]
+            end
```

Comment:
This new code is not tested. We need to add test cases in `spec/datadog/core/configuration/option_spec.rb`

Reply:
Done